support/cas-server-support-aup-jdbc/src/main/java/org/apereo/cas/aup/JdbcAcceptableUsagePolicyRepository.java

```diff
@@ -23,23 +28,44 @@
     private static final long serialVersionUID = 1600024683199961892L;
     
     private final transient JdbcTemplate jdbcTemplate;
-    private final String tableName;
+    private final AcceptableUsagePolicyProperties properties;
 
     public JdbcAcceptableUsagePolicyRepository(final TicketRegistrySupport ticketRegistrySupport,
                                                final String aupAttributeName,
                                                final DataSource dataSource,
-                                               final String tableName) {
+                                               final AcceptableUsagePolicyProperties properties) {
         super(ticketRegistrySupport, aupAttributeName);
         this.jdbcTemplate = new JdbcTemplate(dataSource);
-        this.tableName = tableName;
+        this.properties = properties;
     }
 
     @Override
     public boolean submit(final RequestContext requestContext, final Credential credential) {
         try {
-            final String sql = String.format("UPDATE %s SET %s=true WHERE username=?", this.tableName, this.aupAttributeName);
-            LOGGER.debug("Executing update query [{}]", sql);
-            return this.jdbcTemplate.update(sql, credential.getId()) > 0;
+            final AcceptableUsagePolicyProperties.Jdbc jdbc = properties.getJdbc();
+            String aupColumnName = properties.getAupAttributeName();
+            if (StringUtils.isNotBlank(jdbc.getAupColumn())) {
+                aupColumnName = jdbc.getAupColumn();
+            }
+            final String sql = String.format(jdbc.getSqlUpdateAUP(), jdbc.getTableName(), aupColumnName, jdbc.getPrincipalIdColumn());
+            String principalId = credential.getId();
```

Comment:
Please refactor the logic of determining the principal-id into a separate method, possibly with a `protected` scope.

Reply:
Okay